app/ui/shared/widgets/dropdown-menu-btn.jsx

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2016 Mozilla
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+this file except in compliance with the License. You may obtain a copy of the
+License at http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software distributed
+under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+CONDITIONS OF ANY KIND, either express or implied. See the License for the
+specific language governing permissions and limitations under the License.
+*/
+
+import React, { Component, PropTypes } from 'react';
+import PureRenderMixin from 'react-addons-pure-render-mixin';
+import ImmutablePropTypes from 'react-immutable-proptypes';
+import omit from 'lodash/omit';
+
+import Style from '../style';
+import Btn from './btn';
+import SelectionList from './selection-list';
+
+const BUTTON_PRESS_DURATION = 500; // ms
+
+const DROPDOWN_MENU_BUTTON_STYLE = Style.registerStyle({
+  position: 'relative',
+});
+
+const SELECTION_LIST_STYLE = Style.registerStyle({
+  position: 'absolute',
+  top: '100%',
+  left: 0,
+  maxWidth: '50vw',
+  boxShadow: 'var(--theme-default-shadow)',
+});
+
+class DropdownMenuBtn extends Component {
+  constructor(props) {
+    super(props);
+    this.shouldComponentUpdate = PureRenderMixin.shouldComponentUpdate.bind(this);
+
+    this.state = {
+      showSelectionList: false,
+      selectedIndex: -1,
+    };
+  }
+
+  componentDidMount() {
+    window.addEventListener('click', this.handleWindowClick);
+    window.addEventListener('keyup', this.handleWindowKeyUp);
+  }
+
+  handleWindowClick = e => {
```

Comment:
Why would this get costly? Sure I can imagine a world where there's a lot of listeners because we use hundreds of DropdownMenuBtn for some reason, but why would things get costly?